src/shared/redux/slices/nodeInstallSlice.ts

```diff
@@ -26,16 +26,26 @@ const nodeCheckSlice = createSlice({
 export const { checkingInstallation, nodeInstalled, nodeNotInstalled } =
   nodeCheckSlice.actions;
 
-export const selectNodeCheckStatus = (state: RootState) => state.nodeCheck;
+export const selectNodeCheckStatus = (state: RootState) => state.nodeInstall;
 
-export const checkNode = createAsyncActionMain<void>(
-  'checkForNode',
+export const installNodeJs = createAsyncActionMain<void>(
+  'installNode',
   () => async (dispatch) => {
     dispatch(checkingInstallation());
-    if (await checkForNode()) {
+    const isInstalled = await checkForNode().catch(() =>
+      dispatch(nodeNotInstalled())
+    );
+
+    if (isInstalled) {
       dispatch(nodeInstalled());
     } else {
-      dispatch(nodeNotInstalled());
+      installNode()
+        .then(() => {
+          dispatch(nodeInstalled());
+        })
+        .catch(() => {
+          dispatch(nodeNotInstalled());
+        });
     }
   }
 );
```

Comment:
You can utilize theh `try..catch` block here. 

```ts
export const installNodeJs = createAsyncActionMain<void>(
  'installNode',
  () => async (dispatch) => {
    dispatch(checkingInstallation());

    try {
      const installed = await checkForNode();

      if (!installed) await installNode();

      dispatch(nodeInstalled());
    } catch (error: any) {
      dispatch(nodeNotInstalled());
    }
  }
);
```

Reply:
Done.